src/logic/hooks/useCanTxExecute.tsx

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { extractSafeAddress } from 'src/routes/routes'
+import { currentSafe } from '../safe/store/selectors'
+import useGetRecommendedNonce from './useGetRecommendedNonce'
+
+export const calculateCanTxExecute = (
+  currentSafeNonce: number,
+  preApprovingOwner: string,
+  threshold: number,
+  txConfirmations: number,
+  recommendedNonce?: number,
+  isExecution?: boolean, // when executing from the TxList
+): boolean => {
+  if (isExecution) return true
+
+  const isNextTx = recommendedNonce === currentSafeNonce
```

Comment:
You could actually just move this to line 21. Sorry, I didn't see that it is used once in my last review.